### ISSUE TITLE
Add `have_count` matcher for cursors

### ISSUE THREAD

**Issue by jgebal** [utPLSQL/utPLSQL#488]
I just realized that ability to check for number of rows in cursor could be very handy.

When testing cursors, I write tests that compare whole structure of result-set.
Once structure was tested, I'd like to be able to test filters/parameters.
In this scenario I'm only interested in result size, as the content was already tested by other tests.

Currently I have to test this in following way:
```sql
procedure returns_expected_row_count is
  v_actual t_function_type_cursor;
  v_row    t_function_type_rec; 
begin
  --Act
  v_actual := function_returning_cursor();
  loop
    fetch v_actual into v_row;
    exit when v_actual%NOTFOUND;
  end loop;
  --Assert
  ut.expect( v_actual%ROWCOUNT ).to_equal( 10 );
  close v_actual;
end;
```
The above requires:
- Fetching data (row by row or in bulk)
- Declaring specific cursor variable
- Declaring row variable
- Loop with exit condition
In general - lot of coding to get simple count of rows returned by cursor.

Ideally I should be able to test it using this:
```sql
procedure returns_expected_row_count is
  v_actual sys_refcursor;
begin
  --Act
  v_actual := funcrion_returning_cursor();
  --Assert
  ut.expect( v_actual ).to_have_count_of( 10 );
end;
```

Adding new matcher `to_have_count_of` would make those expectations very simple.

For completeness, the matcher could also work for collections. It's not a must have as collections support the `colllection.COUNT`

**Comment by jgebal** [comment]
Alternative solution would to expose a wrapper type for the cursor.
In that case we could "use" cursor a bot more like Object.
Example:
```sql
procedure returns_expected_row_count is
  v_actual sys_refcursor;
begin
  --Act
  v_actual := funcrion_returning_cursor();
  --Assert
  ut.expect( ut_cursor(v_actual).count ).to_equal( 10 );
end;
```

The usage of `ut_cursor` will open doors for new possibilities for operations on cursor data.

Going with `ut_cursor` would not mean that the support for `sys_refcursor` as a expectation value will be dropped. It would only mean that if users want to do more, they will need to package the `sys_refcursor` with `ut_cursor`.

**Comment by jgebal** [comment]
Resolved by #577
